### PR TITLE
Improve non-admin error message in the UI for bad model registries

### PR DESCRIPTION
**Pull request by YuliaKrimerman**
## Description

When a model registry is misconfigured (e.g. bad DB credentials) or not yet ready, non-admin users only have access to the Service, not the ModelRegistry CR and its status. This change uses the **Endpoints** resource for each registry’s Service to detect availability and improves the UI when a registry is unavailable

Unavailable registries stay in the list; only the selected-registry view and selector styling change when one is chosen.

## How Has This Been Tested?
 **Manual**: Run the app (e.g. PatternFly mode with mock BFF or against a cluster where a registry Service has no ready Endpoints), select or navigate to an unavailable registry, and confirm the new message and that “View details” is hidden.

## Merge criteria:
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages
- [x] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/model-registry/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

- [x] The developer has added tests or explained why testing cannot be added.
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/bff/internal/integrations/kubernetes/client.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ type KubernetesClientInterface interface {
 	GetServiceNames(ctx context.Context, namespace string) ([]string, error)
 	GetServiceDetailsByName(ctx context.Context, namespace, serviceName string, serviceType string) (ServiceDetails, error)
 	GetServiceDetails(ctx context.Context, namespace string) ([]ServiceDetails, error)
+	GetServiceEndpoints(ctx context.Context, namespace, serviceName string) (*corev1.Endpoints, error)
```
Comment (manaswinidas) [inline clients/ui/bff/internal/integrations/kubernetes/client.go:25]: **Breaking change**: Adding `GetServiceEndpoints` to `KubernetesClientInterface` will break all existing implementations (including test mocks). Every struct satisfying this interface must now add this method or the build fails. The mock client doesn't appear to implement it — tests will fail to compile.

File: clients/ui/bff/internal/integrations/kubernetes/shared_k8s_client.go [hunk 2]
```diff
@@ -153,6 +153,34 @@ func (kc *SharedClientLogic) GetServiceDetailsByName(sessionCtx context.Context,
 	return *details, nil
 }
 
+func (kc *SharedClientLogic) GetServiceEndpoints(ctx context.Context, namespace, serviceName string) (*corev1.Endpoints, error) {
```
Comment (manaswinidas) [inline clients/ui/bff/internal/integrations/kubernetes/shared_k8s_client.go:159]: **Deprecated API**: `corev1.Endpoints` is deprecated since k8s v1.33 (CI already flagging this). Use `discoveryv1.EndpointSlice` instead to avoid future breakage.

File: clients/ui/bff/internal/repositories/model_registry.go [hunk 3]
```diff
@@ -54,19 +54,36 @@ func (m *ModelRegistryRepository) GetAllModelRegistriesWithMode(sessionCtx conte
 	var registries = []models.ModelRegistryModel{}
 	for _, s := range resources {
 		serverAddress := m.ResolveServerAddress(s.ClusterIP, s.HTTPPort, s.IsHTTPS, s.ExternalAddressRest, isFederatedMode)
+		isAvailable := m.isRegistryAvailable(sessionCtx, client, namespace, s.Name)
```
Comment (manaswinidas) [inline clients/ui/bff/internal/repositories/model_registry.go:58]: **N+1 problem**: Each iteration calls `isRegistryAvailable` → `GetServiceEndpoints`, making one K8s API call per registry. For N registries this is N extra round-trips. Consider a single batch list of all endpoints in the namespace.

File: clients/ui/bff/internal/repositories/model_registry.go [hunk 4]
```diff
@@ -111,6 +128,7 @@ func (m *ModelRegistryRepository) GetModelRegistryWithMode(sessionCtx context.Co
 		DisplayName:   s.DisplayName,
 		ServerAddress: m.ResolveServerAddress(s.ClusterIP, s.HTTPPort, s.IsHTTPS, s.ExternalAddressRest, isFederatedMode),
 		IsHTTPS:       s.IsHTTPS,
+		IsAvailable:   m.isRegistryAvailable(sessionCtx, client, namespace, modelRegistryID),
```
Comment (manaswinidas) [inline clients/ui/bff/internal/repositories/model_registry.go]: **DRY**: This `ModelRegistryModel` construction is nearly identical to `GetAllModelRegistriesWithMode` above (lines 60-67). Extract a helper like `buildModelRegistryModel(s, serverAddress, isAvailable)` to avoid duplicating field mappings in two places.

File: clients/ui/bff/internal/repositories/model_registry_test.go [hunk 5]
```diff
@@ -26,8 +26,8 @@ var _ = Describe("TestFetchAllModelRegistry", func() {
 
 			By("should match the expected model registries")
 			expectedRegistries := []models.ModelRegistryModel{
-				{Name: "model-registry", Description: "Model Registry Description", DisplayName: "Model Registry", ServerAddress: "http://127.0.0.1:8080/api/model_registry/v1alpha3"},
-				{Name: "model-registry-one", Description: "Model Registry One description", DisplayName: "Model Registry One", ServerAddress: "http://127.0.0.1:8080/api/model_registry/v1alpha3"},
+				{Name: "model-registry", Description: "Model Registry Description", DisplayName: "Model Registry", ServerAddress: "http://127.0.0.1:8080/api/model_registry/v1alpha3", IsAvailable: false},
```
Comment (manaswinidas) [inline clients/ui/bff/internal/repositories/model_registry_test.go]: **Missing coverage**: Every test expects `IsAvailable: false` (the bool zero-value), which only passes because the mock client doesn't implement `GetServiceEndpoints` and `isRegistryAvailable` returns false on error. There are no tests verifying the `IsAvailable: true` path — consider adding a mock that returns valid endpoints.

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelRegistrySelector.tsx [hunk 6]
```diff
@@ -156,7 +156,7 @@ const ModelRegistrySelector: React.FC<ModelRegistrySelectorProps> = ({
         <Bullseye>Model registry</Bullseye>
       </FlexItem>
       <FlexItem>{selector}</FlexItem>
-      {selection && (
+      {selection && selection.isAvailable !== false && (
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelRegistrySelector.tsx]: **DRY / code smell**: The `isAvailable === false` / `!== false` check appears in three different files (here, `ModelRegistrySelectorNavigator.tsx`, and `ModelRegistryCoreLoader.tsx`). Extract a shared helper like `isRegistryUnavailable(mr)` to centralize the backward-compat semantics and the "explicit false only" rule.

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/UnavailableModelRegistry.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        ) : (
+          <WhosMyAdministrator
+            buttonLabel="Who's my administrator?"
+            headerContent="Who's my administrator?"
+            leadText="To request access to a new or existing model registry, contact your administrator."
+            contentTestId="whos-my-admin-content"
+            linkTestId="whos-my-admin-link"
+            popoverPosition={PopoverPosition.left}
+          />
+        )}
```
Comment (manaswinidas) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/UnavailableModelRegistry.tsx]: **DRY**: The `isMUITheme ? <KubeflowDocs .../> : <WhosMyAdministrator .../>` pattern is now repeated in at least three components (`UnavailableModelRegistry`, `ModelRegistryCoreLoader`/`EmptyModelRegistryState`, `ModelRegistrySelector`). Consider extracting a shared `<AdminHelpAction />` component.

**Review by manaswinidas — COMMENTED** [review]
Thanks for the PR! Left inline comments on the high/medium issues — 2 high (interface breaking change, deprecated API) and 5 medium (N+1 perf, DRY violations, missing test coverage).

**Comment by YuliaKrimerman** [comment]
@mturley ready for re-review

File: clients/ui/bff/internal/integrations/kubernetes/client.go [hunk 8]
```diff
@@ -20,6 +20,8 @@ type KubernetesClientInterface interface {
 	GetServiceNames(ctx context.Context, namespace string) ([]string, error)
 	GetServiceDetailsByName(ctx context.Context, namespace, serviceName string, serviceType string) (ServiceDetails, error)
 	GetServiceDetails(ctx context.Context, namespace string) ([]ServiceDetails, error)
+	// GetServiceEndpoints returns the Endpoints for a service (used to detect availability). ODH operator grants RBAC for this. On permission error callers should treat the registry as available.
+	GetServiceEndpoints(ctx context.Context, namespace, serviceName string) (*corev1.Endpoints, error)
```
Comment (mturley) [inline clients/ui/bff/internal/integrations/kubernetes/client.go:26]: The `build` CI check is failing because `staticcheck` flags `corev1.Endpoints` as deprecated (SA1019, deprecated in k8s v1.33+). This hits 4 locations:

1. `clients/ui/bff/internal/integrations/kubernetes/client.go:24` — interface definition
2. `clients/ui/bff/internal/integrations/kubernetes/shared_k8s_client.go:157` — `GetServiceEndpoints` implementation
3. `clients/ui/bff/internal/integrations/kubernetes/shared_k8s_client.go:173` — `EndpointsHasReadyAddresses` helper
4. `clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go:738` — `corev1.EndpointSubset` in test mock

Since we intentionally chose `Endpoints` over `EndpointSlices` for RBAC compatibility (the ODH operator already grants permission for `Endpoints`), please add `//nolint:staticcheck // intentionally using deprecated corev1.Endpoints for RBAC compatibility; see tech debt ticket for EndpointSlice migration` on each of these lines to unblock CI.

File: clients/ui/bff/internal/integrations/kubernetes/client.go [hunk 9]
```diff
@@ -20,6 +20,8 @@ type KubernetesClientInterface interface {
 	GetServiceNames(ctx context.Context, namespace string) ([]string, error)
 	GetServiceDetailsByName(ctx context.Context, namespace, serviceName string, serviceType string) (ServiceDetails, error)
 	GetServiceDetails(ctx context.Context, namespace string) ([]ServiceDetails, error)
+	// GetServiceEndpoints returns the Endpoints for a service (used to detect availability). ODH operator grants RBAC for this. On permission error callers should treat the registry as available.
```
Comment (mturley) [inline clients/ui/bff/internal/integrations/kubernetes/client.go]: This comment and several others reference ODH, which is a downstream distribution. Since this is the upstream kubeflow/model-registry repo, these references should be removed or rephrased to be distribution-agnostic. The 5 occurrences are:

1. `client.go:23` — `"ODH operator grants RBAC for this."`
2. `shared_k8s_client.go:156` — `"ODH operator grants RBAC for Endpoints."`
3. `k8mocks/base_testenv.go:728` — `"(ODH operator grants RBAC for Endpoints)"`
4. `model_registry.go:54` — `"ODH operator grants get-by-name."`
5. `model_registry.go:66` — `"ODH operator grants Endpoints access."`

These could be rephrased to something like "The platform operator is expected to grant RBAC for Endpoints" or simply removed — the code behavior (assume available on permission error) is self-explanatory with the existing non-ODH parts of the comments.
